### ISSUE TITLE
[iceprod2] fix task dependency generation (Trac #1505)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1497]
The current dependency system is broken and won't work.

A better solution from the new docs I just wrote up:

```text
Dataset Submission
------------------

The submit process takes a dataset config as input, usually
through the website. It then creates the following entries:

* dataset
* config
* task_rel - one for each task

Note that if the IceProd instance you submit through does not
know about a dataset dependency, submission will fail. Try submitting
through the master instance in that case.

Buffering
---------

Jobs and tasks get buffered, or allocated, when there is empty
space on a queue that they can run at. Whole jobs are buffered at
once, so more than the requested number of tasks may be buffered.

The following entries are made:

* search
* task
* job
```

Migrated from https://code.icecube.wisc.edu/ticket/1505
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:28", 
    "description": "The current dependency system is broken and won't work.\n\nA better solution from the new docs I just wrote up:\n\n{{{\nDataset Submission\n------------------\n\nThe submit process takes a dataset config as input, usually\nthrough the website. It then creates the following entries:\n\n* dataset\n* config\n* task_rel - one for each task\n\nNote that if the IceProd instance you submit through does not\nknow about a dataset dependency, submission will fail. Try submitting\nthrough the master instance in that case.\n\nBuffering\n---------\n\nJobs and tasks get buffered, or allocated, when there is empty\nspace on a queue that they can run at. Whole jobs are buffered at\nonce, so more than the requested number of tasks may be buffered.\n\nThe following entries are made:\n\n* search\n* task\n* job\n}}}", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067088921308", 
    "component": "iceprod", 
    "summary": "[iceprod2] fix task dependency generation", 
    "priority": "blocker", 
    "keywords": "", 
    "time": "2016-01-11T20:35:56", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160323T20:43:20: 
* david.schultz changed status from "new" to "closed"
* david.schultz changed resolution from "" to "fixed"
* david.schultz commented:

> In changeset 143667:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="143667"
> new task dependency system. fixes #1497
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:28: 
* olivas changed milestone from "IceProd 2.0" to ""
* olivas commented:

> Milestone IceProd 2.0 deleted
